src/thunderbots/software/ai/hl/stp/play/corner_kick_play.cpp

```diff
@@ -0,0 +1,216 @@
+#include "ai/hl/stp/play/corner_kick_play.h"
+
+#include <ai/hl/stp/evaluation/ball.h>
+
+#include <g3log/g3log.hpp>
+
+#include "ai/hl/stp/evaluation/ball.h"
+#include "ai/hl/stp/evaluation/possession.h"
+#include "ai/hl/stp/play/play_factory.h"
+#include "ai/hl/stp/tactic/move_tactic.h"
+#include "ai/hl/stp/tactic/passer_tactic.h"
+#include "ai/hl/stp/tactic/receiver_tactic.h"
+#include "ai/passing/pass_generator.h"
+#include "shared/constants.h"
+#include "util/logger/custom_logging_levels.h"
+
+using namespace Passing;
+
+const std::string CornerKickPlay::name = "Corner Kick Play";
+
+CornerKickPlay::CornerKickPlay() : MAX_TIME_TO_COMMIT_TO_PASS(Duration::fromSeconds(5)) {}
+
+std::string CornerKickPlay::getName() const
+{
+    return CornerKickPlay::name;
+}
+
+bool CornerKickPlay::isApplicable(const World &world) const
+{
+    return world.gameState().isOurDirectFree() &&
+           Evaluation::ballInFriendlyCorner(world.field(), world.ball(),
+                                            BALL_IN_CORNER_RADIUS);
+}
+
+bool CornerKickPlay::invariantHolds(const World &world) const
+{
+    return !Evaluation::teamHasPossession(world.enemyTeam(), world.ball());
+}
+
+void CornerKickPlay::getNextTactics(TacticCoroutine::push_type &yield)
+{
+    // TODO: Rough description of what this function does (stages, tactics, etc.) here
+
+    // Figure out if we're taking the kick from the +y or -y corner
+    bool kick_from_pos_corner = world.ball().position().y() > 0;
+
+    // We want the two cherry pickers to be in rectangles on the +y and -y sides of the
+    // field in the +x half. We also further offset the rectangle from the goal line
+    // for the cherry-picker closer to where we're taking the corner kick from
+    Vector enemy_defense_area_pos_y_x_offset(world.field().enemyDefenseArea().width(), 0);
```

Comment:
Maybe `pos_y_goalline_offset` is a better name here? This offset isn't really related to the defense area other than that's related to its value. It's an offset from the goalline, so I think this name is a bit more descriptive.

Reply:
changed to `pos_y_goalline_x_offset`